smelt.lic

```diff
@@ -25,80 +17,89 @@ class Smelt
 
     args = parse_args(arg_definitions)
 
-    get_crafting_item(@rod, @bag, @bag_items, @belt)
+    @settings = get_settings
+    @bag = @settings.crafting_container
+    @bag_items = @settings.crafting_items_in_container
+    @belt = @settings.forging_belt
+    @rod = @settings.forging_tools.find { |item| /rod/ =~ item }
+    @adjustable_tongs = @settings.adjustable_tongs
+
+    get(@rod)
     if args.refine
-      refine
+      get('flux')
+      type("pour", "flux", "in crucible", "flux", true)
     else
-      stir
+      type("stir", "crucible", "with my rod", "rod", false)
+      work
+      stow(@rod)
     end
-    stow_crafting_item(@rod, @bag, @belt)
   end
 
-  def refine
-    bput('get flux', 'you get')
-    case bput('pour flux in crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'down and needs more fuel', 'roundtime')
-    when /roundtime/i
-      waitrt?
-      fput('stow flux')
-      return stir
-    when 'clumps of molten metal'
-      waitrt?
-      fput('stow flux')
-      return turn
-    when 'flickers and is unable to consume'
-      waitrt?
-      fput('stow flux')
-      return bellows
-    when 'down and needs more fuel'
-      waitrt?
-      fput('stow flux')
-      return fuel
-    end
+  def get(name)
+    DRCC.get_crafting_item(name, @bag, @bag_items, @belt)
   end
 
-  def stir
-    pause 1
-    waitrt?
-    case bput("stir crucible with my #{@rod}", 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
-    when /roundtime/i
-      return stir
-    when 'You can only mix a crucible'
-      return
-    when 'clumps of molten metal'
-      return turn
-    when 'flickers and is unable to consume'
-      return bellows
-    when 'needs more fuel', 'needs some more fuel'
-      return fuel
-    end
+  def stow(name)
+    DRCC.stow_crafting_item(name, @bag, @belt)
   end
 
-  def turn
-    pause 1
-    waitrt?
-    bput('turn crucible', 'roundtime')
-    stir
+  def action(verb, object, tool)
+    DRC.bput("#{verb} #{object} #{tool}", 'Pour what', 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
```

Comment:
Since the args are actually instance vars, you could just remove them:
```suggestion
  def action()
    DRC.bput("#{@verb} #{@object} #{@tool}" ......
```